sdk/storage/Azure.Storage.Files/src/Models/StorageFileProperties.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for
+// license information.
+
+using System;
+using System.Collections.Generic;
+using Azure.Core.Http;
+
+namespace Azure.Storage.Files.Models
+{
+    /// <summary>
+    /// Properties for a file.
+    /// </summary>
+    public class StorageFileProperties
+    {
+        /// <summary>
+        /// The internal RawStorageFileProperties
+        /// </summary>
+        internal RawStorageFileProperties _rawStorageFileProperties;
+
+        /// <summary>
+        /// The DateTimeOffset when the file was last modified.
+        /// </summary>
+        public DateTimeOffset LastModified => this._rawStorageFileProperties.LastModified;
+
+        /// <summary>
+        /// A set of name-value pairs associated with this file as user-defined metadata.
+        /// </summary>
+        public IDictionary<string, string> Metadata => this._rawStorageFileProperties.Metadata;
+
+        /// <summary>
+        /// Returns the type File. Reserved for future use.
+        /// </summary>
+        public Header FileType => this._rawStorageFileProperties.FileType;
```

Comment:
What's this `Header` type and should we consider renaming it to avoid conflicts with a common term?

Reply:
I just copied it up for the generated type, I don't have any more information about it.

Reply:
The docs say it's only got the value `File` and reserved for future use.  I think we should include this on the thread with the service folks to figure out a better name for this than `Header`.

Reply:
@tg-msft will do. thanks

Reply:
Filed #7389 to track this.